server/new-commiter.go

```diff
@@ -0,0 +1,163 @@
+			StatusCode: http.StatusBadRequest,
+			Where:      "p.ExecuteCommand",
+		}
+	}
+
+	attachments := []*model.SlackAttachment{{
+		Title:      "Fetching new commiters since " + since.Format(shortFormWithDay),
+		Text:       waitText,
+		AuthorName: organisation,
+		AuthorIcon: org.GetAvatarURL(),
+		AuthorLink: fmt.Sprintf("https://github.com/%v", organisation),
+	}}
+
+	loadingPost := &model.Post{
+		ChannelId: args.ChannelId,
+		UserId:    p.botUserID,
+	}
+	model.ParseSlackAttachment(loadingPost, attachments)
+
+	loadingPost, appErr := p.API.CreatePost(loadingPost)
+	if appErr != nil {
+		return appErr
+	}
+
+	go p.updateNewCommitersPost(loadingPost, args.UserId, organisation, since)
+
+	return nil
+}
+
+func (p *Plugin) updateNewCommitersPost(post *model.Post, userID, org string, since time.Time) {
+	contributors, err := p.fetchContributorsFromOrg(org)
+	if err != nil {
+		p.API.LogError("failed to fetch data", "err", err.Error())
+
+		var message string
+		if _, ok := err.(*github.RateLimitError); ok {
+			message = "Hit rate limit. Please try again later."
+		} else {
+			message = "Failed to fetch data:" + err.Error()
+		}
+		post.Props["attachments"].([]*model.SlackAttachment)[0].Text = message
+	} else {
+		firstContributions := p.findFirstContributions(contributors)
+		p.filterContributionsByWeek(firstContributions, since)
+
+		var result []*firstContributionInfo
+		for commiter, contribution := range firstContributions {
+			commits, _ := p.fetchCommitsFromRepoByAuthorAndWeek(org, contribution.repo, commiter, contribution.contributionWeek)
+			firstCommit := commits[0]
+			if firstCommit.GetCommit().Committer.GetDate().After(since) {
+				result = append(result, &firstContributionInfo{firstCommit.GetAuthor().GetLogin(), firstCommit.GetCommit().Committer.GetDate(), firstCommit.GetHTMLURL()})
+			}
+		}
+
+		sort.Slice(result, func(i, j int) bool {
+			return result[i].date.Before(result[j].date)
+		})
+
+		var resultText string
+		for _, e := range result {
+			resultText += fmt.Sprintf("- [%[1]s](https://github.com/%[1]v)  [first commit](%s)  %s\n", e.author, e.commit, e.date.Format(shortFormWithDay))
+		}
+
+		attachment := post.Props["attachments"].([]*model.SlackAttachment)[0]
+		attachment.Title = "New Commiters since " + since.Format(shortFormWithDay)
+		attachment.Text = ""
+		attachment.Fields = []*model.SlackAttachmentField{{
+			Title: "Number of new commiters",
+			Value: strconv.Itoa(len(result)),
+		}, {
+			Title: "Commiters and their first commits",
+			Value: resultText,
+		}}
+	}
+
+	if _, appErr := p.API.UpdatePost(post); appErr != nil {
+		p.SendEphemeralPost(post.ChannelId, userID, "Something went bad. Please try again. "+appErr.Where+" "+appErr.Id)
+		p.API.LogError("failed to update post", "err", appErr.Error())
+		return
+	}
+}
+
+func (p *Plugin) findFirstContributions(contributors []*repoContributors) map[string]contributionInfo {
+	firstContributions := map[string]contributionInfo{}
```

Comment:
```suggestion
	contributions := map[string]contributionInfo{}
```
suggest shortening the name since the function name already implies "first" contributions

Reply:
`firstContributions` makes it quite clear to me what it contains and helped me understand the code. 2/5 to not change it.

Reply:
fair enough..  Please omit this change request and leave as is :)